internal/autoevent/executor.go

```diff
@@ -0,0 +1,94 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2019 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package autoevent
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/handler"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+)
+
+const millionSecPerDay uint32 = 86400000
+
+type Executor interface {
+	Run()
+	Stop()
+}
+
+type executor struct {
+	deviceName   string
+	autoEvent    models.AutoEvent
+	lastReadings map[string]string
```

Comment:
Note, you'll also need a separate map for binary readings.

Reply:
we could create another PR after CBOR implementation completed